### PR TITLE
47ified alliance

### PR DESCRIPTION
**Pull request by faddat**
This PR upgrades alliance to cosmos-sdk v0.47.0

There is a notion going about that 47 could be better than 46, I'm generally bought into this.  So this PR might also smooth out some upgrades. 

Since alliance is a module, we should likely have a 46'd alliance, as well as a 47'd alliance.

### CODE REVIEW

File: custom/bank/keeper/keeper.go [hunk 1]
```diff
@@ -35,9 +35,10 @@ func NewBaseKeeper(
 	ak accountkeeper.AccountKeeper,
 	paramSpace paramtypes.Subspace,
 	blockedAddrs map[string]bool,
+	authority string,
 ) Keeper {
 	keeper := Keeper{
-		BaseKeeper: bankkeeper.NewBaseKeeper(cdc, storeKey, ak, paramSpace, blockedAddrs),
+		BaseKeeper: bankkeeper.NewBaseKeeper(cdc, storeKey, ak, blockedAddrs, authority), // TODO: how to set authority?
```
Comment (faddat) [inline custom/bank/keeper/keeper.go:41]: note for review.  Will be checking how upstream does this tomorrow.

File: proto/alliance/events.proto [hunk 2]
```diff
@@ -1,5 +1,5 @@
 syntax = "proto3";
-package alliance.alliance;
+package alliance;
```
Comment (faddat) [inline proto/alliance/events.proto:2]: @ThanhNhann why'd the name of the proto package change here?

Comment (faddat) [reply proto/alliance/events.proto:2]: (and in the other .proto files)

Comment (ThanhNhann) [reply proto/alliance/events.proto:2]: I see that Alliance's struct proto folder doesn't have submodule folder so I think it's correct to change it like now

Comment (faddat) [reply proto/alliance/events.proto:2]: Nice.  Can you confirm @javiersuweijie ?

Comment (javiersuweijie) [reply proto/alliance/events.proto:2]: Seems correct but I'll need to check if this is a breaking change that updates the GRPC types for the alliance models.

**Comment by faddat** [comment]
@javiersuweijie sir, thanks so much for the review and fixes!

**Pull request merged**